crates/egui-winit/src/lib.rs

```diff
@@ -820,11 +821,16 @@ impl State {
 
         if let Some(ime) = ime {
             let pixels_per_point = pixels_per_point(&self.egui_ctx, window);
-            let ime_rect_px = pixels_per_point * ime.rect;
-            if self.ime_rect_px != Some(ime_rect_px)
-                || self.egui_ctx.input(|i| !i.events.is_empty())
-            {
-                self.ime_rect_px = Some(ime_rect_px);
+            let ime_rect_px = ime.rect * pixels_per_point;
+            let mut need_set_ime_cursor_area = true;
+
+            // On Wayland of Linux, repaints every frame Issue : See https://github.com/emilk/egui/pull/4254
+            if self.egui_ctx.os() == egui::os::OperatingSystem::Nix {
+                need_set_ime_cursor_area = self.ime_rect_px != Some(ime_rect_px)
+                    && self.egui_ctx.input(|i| !i.events.is_empty());
```

Comment:
Do we need to treat `Nix` specially here? Isn't this logic sound for all OSes?

Reply:
I think it should only be applied on Linux.
Definitely shouldn't be applied on Windows.